thirdparty/cmdconfig/commands/cmdeval/cmdeval.go

```diff
@@ -102,14 +109,71 @@ type EvalFnRunner struct {
 	IncludeMetaResources bool
 	Ctx                  context.Context
 	Selector             kptfile.Selector
+	dataItems            []string
 }
 
 func (r *EvalFnRunner) runE(c *cobra.Command, _ []string) error {
 	err := runner.HandleError(r.Ctx, r.RunFns.Execute())
 	if err != nil {
 		return err
 	}
-	return cmdutil.WriteFnOutput(r.Dest, r.OutContent.String(), r.FromStdin, printer.FromContextOrDie(r.Ctx).OutStream())
+	if err = cmdutil.WriteFnOutput(r.Dest, r.OutContent.String(), r.FromStdin,
+		printer.FromContextOrDie(r.Ctx).OutStream()); err != nil {
+		return err
+	}
+	if r.AddImage {
+		r.AddImageToKptfile()
+	}
+	return nil
+}
+
+// AddImageToKptfile adds the function image and config to Kptfile
+func (r *EvalFnRunner) AddImageToKptfile() {
+	pr := printer.FromContextOrDie(r.Ctx)
+	kptFile, err := pkg.ReadKptfile(filesys.FileSystemOrOnDisk{}, r.Dest)
+	if err != nil {
+		pr.Printf("image not added: Kptfile not exists\n")
+		return
+	}
+	// TODO(yuwenma): Right now we cannot tell a function is a mutator or validator. Once kpt supports
+	// OCI images, we can add annotations to image and find out the function type from these annotations.
+	if kptFile.Pipeline == nil {
+		kptFile.Pipeline = &kptfile.Pipeline{}
+	}
+	if kptFile.Pipeline.Mutators == nil {
+		kptFile.Pipeline.Mutators = []kptfile.Function{}
+	} else {
+		for _, m := range kptFile.Pipeline.Mutators {
+			if m.Name == r.Image || m.Image == r.Image {
+				pr.Printf("skip adding image: already exists in Kptfile\n")
+				return
+			}
+		}
+	}
+	var newMutator kptfile.Function
+	if r.FnConfigPath != "" {
+		newMutator = kptfile.Function{Name: r.Image, Image: r.Image, ConfigPath: r.FnConfigPath}
```

Comment:
One thing we will have to do is: if `fnConfigPath` is absolute, that will have to be converted to relative otherwise it will result in a invalid Pipeline.

Reply:
I see. good catch! Thanks

Reply:
done